src/core/ScreenLockListenerWin.cpp

```diff
@@ -0,0 +1,69 @@
+#include "ScreenLockListenerWin.h"
+#include <QApplication>
+#include <windows.h>
+#include <wtsapi32.h>
+
+/*
+ * See https://msdn.microsoft.com/en-us/library/windows/desktop/aa373196(v=vs.85).aspx
+ * See https://msdn.microsoft.com/en-us/library/aa383841(v=vs.85).aspx
+ * See https://blogs.msdn.microsoft.com/oldnewthing/20060104-50/?p=32783
+ */
+ScreenLockListenerWin::ScreenLockListenerWin(QWidget *parent) :
+    ScreenLockListenerPrivate(parent),
+    QAbstractNativeEventFilter()
+{
+    Q_ASSERT(parent!=NULL);
+    // On windows, we need to register for platform specific messages and
+    // install a message handler for them
+    QCoreApplication::instance()->installNativeEventFilter(this);
+
+    // This call requests a notification from windows when a laptop is closed
+    HPOWERNOTIFY hPnotify = RegisterPowerSettingNotification(
+                reinterpret_cast<HWND>(parent->winId()),
+                &GUID_LIDSWITCH_STATE_CHANGE, DEVICE_NOTIFY_WINDOW_HANDLE);
+    m_powernotificationhandle = reinterpret_cast<void*>(hPnotify);
+
+    // This call requests a notification for session changes
+    if (!WTSRegisterSessionNotification(
+                reinterpret_cast<HWND>(parent->winId()),
+                NOTIFY_FOR_THIS_SESSION)){
+    }
+}
+
+ScreenLockListenerWin::~ScreenLockListenerWin()
+{
+    HWND h= reinterpret_cast<HWND>(static_cast<QWidget*>(parent())->winId());
+    WTSUnRegisterSessionNotification(h);
+
+    if(m_powernotificationhandle){
```

Comment:
Coding style, please add a space after if. Also applies to the following if conditions in this file.